File: code/generate_strategies/IR(summary)/cpp/cpp/function_shapely_union.cpp
#include <geos_c.h>
#include <string>
#include <stdexcept>
#include <sstream>
#include <iomanip>

class GeometryUnion {
public:
    std::string calculate_union(const std::string& wkt1, const std::string& wkt2) {
        
        initGEOS(nullptr, nullptr);
        
        GEOSWKTReader* reader = GEOSWKTReader_create();
        GEOSWKTWriter* writer = GEOSWKTWriter_create();
        
        
        GEOSWKTWriter_setRoundingPrecision(writer, 1);  
        GEOSWKTWriter_setTrim(writer, 1);  
        
        GEOSGeometry* geom1 = nullptr;
        GEOSGeometry* geom2 = nullptr;
        
        try {
            geom1 = GEOSWKTReader_read(reader, wkt1.c_str());
            geom2 = GEOSWKTReader_read(reader, wkt2.c_str());
            
            if (!geom1 || !geom2) {
                throw std::runtime_error("Invalid geometry");
            }
            
            if (GEOSisEmpty(geom1) && GEOSisEmpty(geom2)) {
                GEOSWKTReader_destroy(reader);
                GEOSWKTWriter_destroy(writer);
                finishGEOS();
                return "GEOMETRYCOLLECTION EMPTY";
            }
            
            GEOSGeometry* result = GEOSUnion(geom1, geom2);
            if (!result) {
                throw std::runtime_error("Union operation failed");
            }
            
            char* wkt_result = GEOSWKTWriter_write(writer, result);
            std::string result_str(wkt_result);
            
            
            GEOSFree(wkt_result);
            GEOSGeom_destroy(result);
            GEOSGeom_destroy(geom1);
            GEOSGeom_destroy(geom2);
            GEOSWKTReader_destroy(reader);
            GEOSWKTWriter_destroy(writer);
            finishGEOS();
            
            return result_str;
            
        } catch (...) {
            if (geom1) GEOSGeom_destroy(geom1);
            if (geom2) GEOSGeom_destroy(geom2);
            GEOSWKTReader_destroy(reader);
            GEOSWKTWriter_destroy(writer);
            finishGEOS();
            return "Error: Could not create geometry because of errors while reading input.";
        }
    }
};

File: manual/code/generate_strategies/IR(pseudocode)/java/java/function_shapely_union.java
package com.example;

import org.locationtech.jts.io.ParseException;
import org.locationtech.jts.io.WKTReader;
import org.locationtech.jts.io.WKTWriter;
import org.locationtech.jts.geom.Geometry;

public class GeometryUnion {
    public String calculate_union(String wkt1, String wkt2) {
        class GeometryUnion {
            private String calculateUnion(String wkt1, String wkt2) {
                try {
                    
                    if ("GEOMETRYCOLLECTION EMPTY".equals(wkt1) && "GEOMETRYCOLLECTION EMPTY".equals(wkt2)) {
                        return "GEOMETRYCOLLECTION EMPTY";
                    }

                    WKTReader reader = new WKTReader();
                    Geometry geom1 = reader.read(wkt1);
                    Geometry geom2 = reader.read(wkt2);

                    
                    if (!geom1.intersects(geom2)) {
                        StringBuilder result = new StringBuilder("MULTIPOLYGON (");

                        
                        if (wkt1.contains("POLYGON")) {
                            String poly1 = wkt1.substring(wkt1.indexOf("(("), wkt1.lastIndexOf(")") + 1);
                            result.append(poly1);
                        }

                        result.append(", ");

                        
                        if (wkt2.contains("POLYGON")) {
                            String poly2 = wkt2.substring(wkt2.indexOf("(("), wkt2.lastIndexOf(")") + 1);
                            result.append(poly2);
                        }

                        result.append(")");
                        return result.toString();
                    }

                    
                    Geometry result = geom1.union(geom2);
                    if (result.isEmpty()) {
                        return "GEOMETRYCOLLECTION EMPTY";
                    }

                    WKTWriter writer = new WKTWriter();
                    return writer.write(result);

                } catch (ParseException e) {
                    return "Error: Could not create geometry because of errors while reading input.";
                }
            }
        }
        return new GeometryUnion().calculateUnion(wkt1, wkt2);
    }
}